Weekly Updates: Fix crash caused by modifying the UI from a background thread

hassaanelgarem opened the pull request:
Fixes #20266

## Description
The crash is caused by calling `JetpackNotificationMigrationService.shared.shouldPresentNotifications()` from a background thread. This gets triggered by the weekly roundup Background Processing task. This PR fixes this crash by avoiding the call to `RootViewCoordinator.shared` when assessing if JP features are enabled or not from a background thread.

## Testing Instructions

1. Checkout `trunk`
2. Perform the reproduction steps described in https://github.com/wordpress-mobile/WordPress-iOS/issues/20266#issuecomment-1458394093
3. Observe that the app crashes with the same stack trace as the original crash
4. Checkout this branch
5. Perform the same steps again
6. Observe that the app doesn't crash

P.S: If the app is left running, it might crash due to the workaround done to reproduce the crash. The workaround results in us repeatedly initializing `RootViewCoordinator`, which can eventually lead to a different crash.

## Regression Notes
1. Potential unintended areas of impact
Jetpack features removal

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested going back and forth from the normal phase,  phase 4, and new users phase. 

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

hassaanelgarem committed:
Fix: avoid initializing `RootViewCoordinator` from a background thread
when checking if JP features are enabled

hassaanelgarem committed:
Add: release note

hassaanelgarem commented:
@mokagio FYI 🙇

momo-ozawa reviewed (approved):
Works as described 👍

mokagio committed:
Move release notes entry for #20275 to version 21.8.1

The fix in #20275 was originally made on top of the `release/21.9`
branch but we decided to ship it as a hotfix.

I could have amended 9dfe3379997faab2bab14698101959b5c3d814d1 with this
change, but decided to keep it in one piece to preserve a record of the
original change.

mokagio commented:
@hassaanelgarem @momo-ozawa just a note to say that I rewrote the history on this branch in order to base it on top of the new `release/21.8.1` hotifx. 

This is something I discussed with @hypest [just a few hours ago](https://github.com/wordpress-mobile/WordPress-iOS/pull/20214#issuecomment-1458077193) and the intent is to preserve attribution. With this approach, the PR that will land in the hotfix branch will have been created by @hassaanelgarem as opposed to the usual ad-hoc PR I would create to cherry pick.

This is the first time I try this approach and, apart from the rewriting of history of someone else's branch which always feels odd, I think it's working alright. Keen to know what you think.

For review reference, I pushed a backup of the original history. We can use it to compare [the diff of that version against `release/21.9`](https://github.com/wordpress-mobile/WordPress-iOS/compare/release/21.9...backup/fix/20266-weekly-updates-background-thread-crash) and this one to ensure the match (aside from the `RELEASE-NOTES.txt` version header change).

mokagio commented:
I'll merge this as soon as CI gives the okay then move on with the hotfix deployment.

For the record, I verified the fix behavior on top of 21.8.1 via:

1. Checked out 21.8.1 and applied the code changes to reproduce the crash, bfb0c805dd0ac732e087ba6d9f4f9d5baeac01f3
2. Reproduced the crash
3. Checked out this branch and applied  the code changes to reproduce the crash
4. Verified the crash doesn't occur and that `shouldPresentNotifications` is called (via breakpoint)
